libraries/AP_GPS/AP_GPS_PX4.cpp

```diff
@@ -39,9 +39,9 @@ AP_GPS_PX4::AP_GPS_PX4(AP_GPS &_gps, AP_GPS::GPS_State &_state, AP_HAL::UARTDriv
 }
 
 
-#define MS_PER_WEEK             7*24*3600*1000LL
-#define DELTA_POSIX_GPS_EPOCH   315964800LL*1000LL
-#define LEAP_MS_GPS_2014        16*1000LL
+const uint64_t MS_PER_WEEK            = ((uint64_t)7)*24*3600*1000;
+const uint64_t DELTA_POSIX_GPS_EPOCH  = ((uint64_t)3657)*24*3600*1000;
+const uint64_t LEAP_MS_GPS_2016       = ((uint64_t)17)*1000;
```

Comment:
The true number of leap seconds is currently not available to the native PX4 driver for the Zubax GNSS. The AP_GPS_PX4 driver in turn is only a very thin interface conversion layer for APM, so I currently don't see a way to change this. 

If this is mission critical problem, we should probably file a feature request in PX4/Firmware and/or in the Zubax/gnss project.